### ISSUE TITLE
Trigger change on button click

### ISSUE THREAD

**Issue by rajkaran** [fcorti/adf-pentaho-project#2]
I have a usecase where I have four fields and a button. As shown in below screenshot

![r](https://user-images.githubusercontent.com/4390019/35392703-b5e7194e-01b0-11e8-9d40-b2d674318aaf.png)

Currently we can send params to dashboard but changing multiple fields at a time will cause dashboard to load multiple times.

Can we have a way to hold the load until button is click.

**Comment by fcorti** [comment]
It should be doable... but before I need to understand how to do it into the `requirejs` function.
The first thing to do will be to remove the `fireChange` setup in the javascript source code.
What I have to experiment is how to trigger the refresh of the dashboard when the button is pressed.
I need some experimentation here but If someone have any idea of the command to refresh dashboard, please share to save some effort. ;-)

**Comment by fcorti** [comment]
Maybe this can help:
https://help.pentaho.com/Documentation/8.0/Developer_Center/CDF_API

**Comment by rajkaran** [comment]
I have an idea which will not require any change to project.

Have a button which will change value of  a parameter and only have this parameter in dashboard's listener list.

When user change the field's values through firechange parameters will get updated but table won't load until last parameter doesn't change. On button click we can change it's value to any random number.

**Comment by rajkaran** [comment]
I have implemented the solution I have mentioned before. It is a dirty solution I know but its one possible solution.

Below is the code.
Component
```
import { Component, OnInit, ViewChild, ElementRef, TemplateRef, Renderer } from '@angular/core';
import { Router } from '@angular/router';
import { PentahoDashboardService } from 'pentaho-dashboard';

@Component({
    selector: 'app-parent-dashboard',
    templateUrl: './parent-dashboard.component.html',
    styleUrls: ['./parent-dashboard.component.css']
})

export class ParentDashboardComponent implements OnInit {

    from = '2018-01-02 00:00:00';
    to = '2018-01-08 00:00:00';
    groupBy = 'day';
    physician = 'Pederson';
    triggerLoad = Math.random();

    @ViewChild('buttonClickEvent') buttonClickEvent: ElementRef;

    constructor(
        private router: Router,
        private pentahoDashboardService: PentahoDashboardService,
        private renderer: Renderer
    ) {
        if (this.pentahoDashboardService.isNotLoggedIn()) {
            this.router.navigate(['/login']);
        }
    }

    ngOnInit() {
    }

    changeEventValue(){
        let event = new CustomEvent('change', {bubbles: true});
        this.renderer.invokeElementMethod(
            this.buttonClickEvent.nativeElement, 'dispatchEvent', [event]
        );

        this.triggerLoad = Math.floor((Math.random() * 1000) + 1);
    }

}

```
Template
```
<div>

    <input [(ngModel)]="from" placeholder="FROM" id="from-datetime">
    <input [(ngModel)]="to" placeholder="TO" id="to-datetime">
    <input [(ngModel)]="groupBy" placeholder="GROUP" id="group-by">
    <input [(ngModel)]="physician" placeholder="PHYSICIAN" id="physician-lastName">
    <input type = "hidden" [(ngModel)]="triggerLoad" placeholder="A randomly generated number" id="button-click-event" #buttonClickEvent>
    <input type="button" (click)="changeEventValue()" value="Generate Report">


    <pentaho-dashboard
        id = "parent"
        pentahoPath = "/public/basic-cde-samples/Embed PRC patient visit.wcdf"
        [params] = "['fromDatetime', 'toDatetime', 'groupBy', 'physicianLastName', 'button_click_event']"
        [masterHtmlElementIds] = "['from-datetime', 'to-datetime', 'group-by', 'physician-lastName', 'button-click-event']"
        setDefaults = "true" >
    </pentaho-dashboard>
</div>

```

**Comment by fcorti** [comment]
Ok, now also this is included in release 1.0.5.
The documentation is fully updated.
